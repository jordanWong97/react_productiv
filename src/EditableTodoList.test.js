import React from "react";
import { render } from "@testing-library/react";
import EditableTodoList from "./EditableTodoList";

describe("productiv app", function () {
  it("renders without crashing", function () {
    render(<EditableTodoList todos={[{
      id: 1,
      title: "testTitle",
      description: "testDesc",
      priority: 1
    }]} />);
  });

  it("contains an edit button for a todo", function () {
    const { queryByText } = render(<EditableTodoList todos={[{
      id: 1,
      title: "testTitle",
      description: "testDesc",
      priority: 1
    }]} />);
    expect(queryByText("Edit")).toBeInTheDocument();
  });

  it("renders title of an editable todo", function () {
    const { queryByText } = render(<EditableTodoList todos={[{
      id: 1,
      title: "testTitle",
      description: "testDesc",
      priority: 1
    }]} />);
    expect(queryByText("testTitle")).toBeInTheDocument();
  });

  it("snapshot test", function () {
    const { container } = render(<EditableTodoList todos={[{
      id: 1,
      title: "testTitle",
      description: "testDesc",
      priority: 1
    }]} />);
    expect(container).toMatchSnapshot();
  });
});

