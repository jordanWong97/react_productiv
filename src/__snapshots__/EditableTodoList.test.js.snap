// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`productiv app snapshot test 1`] = `
<div>
  <div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
        >
          <div>
            <b>
              testTitle
            </b>
             
            <small>
              (priority: 
              1
              )
            </small>
          </div>
          <div>
            <small>
              testDesc
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
